vortex-dtype/src/dtype.rs

```diff
@@ -51,7 +51,10 @@ impl DType {
             Primitive(_, n) => matches!(n, Nullable),
             Utf8(n) => matches!(n, Nullable),
             Binary(n) => matches!(n, Nullable),
-            Struct(st, _) => st.dtypes().iter().all(DType::is_nullable),
+            Struct(st, n) => {
+                matches!(n, Nullable)
+                    || !st.dtypes.is_empty() && st.dtypes().iter().all(DType::is_nullable)
+            }
```

Comment:
This is a fun one, when we added nullability to struct types we didn't check it here. What is worse though `all` neutral element is true which means empty structs were by default nullable